### PR TITLE
Documentation cleanup

### PR DESCRIPTION
**Pull request by rowanc1**
This PR edits the docs around some of the nodes, changes Node --> Parent if the node has children, and changes the rendering of the MyST docs as dd/dl/dt lists instead of bullet points (better styling).

### CODE REVIEW

File: schema/admonitions.schema.json [hunk 1]
```diff
@@ -39,14 +39,15 @@
...
+              },
+              "description": "An optional `admonitionTitle` followed by the admonitions content."
             },
             "position": {},
             "data": {}
           },
           "additionalProperties": false
         },
-        { "$ref": "unist.schema.json#/$defs/Node" }
+        { "$ref": "unist.schema.json#/$defs/Parent" }
```
Comment (fwkoch) [inline schema/admonitions.schema.json:50]: Should we swap this back to `Node` like you did for cross reference? We are already validating `children` above in a more specific way; could cause problems to validate again here if there's some conflict.

**Review by fwkoch — APPROVED** [review]
Love all the better descriptions!

File: schema/admonitions.schema.json (hunk 1 above)
Comment (rowanc1) [reply schema/admonitions.schema.json]: Sounds good, switched back.

**Comment by rowanc1** [comment]
This is now just pure documentation changes, going to hit merge.

**Pull request merged**